### PR TITLE
Fix serialization bug in `BroadcastJoinLayer`

### LINKED ISSUE #9870: Broadcast join fails when passing a list of columns to merge on
**Describe the issue**:

The broadcast merge codepath for dataframes throws an error when passing a list of columns in the `on` argument with a `keyError`

**Minimal Complete Verifiable Example**:

```python
import pandas as pd
import numpy as np
from dask import dataframe as dd
from distributed import Client, wait
c = Client()

df1 = pd.DataFrame({"a":np.arange(20),"b":np.arange(20),"c":[1,2]*10})
df2 = df1.copy(deep=True)
df1 = dd.from_pandas(df1,2)
df2 = dd.from_pandas(df2,5)

len(df1.merge(df2,on=["a"], how="inner", shuffle="tasks", broadcast=True))
# KeyError: "('a',)"
# Works with on="a"
```

**Anything else we need to know?**:
Same issue persists when joining on multiple columns

**Environment**:

- Dask version: 2022.12.0
- Python version: 3.9
- Operating System: ubuntu18.04
- Install method (conda, pip, source): pip

### PR DESCRIPTION
**Pull request by rjzamora**
The current logic used to serialize a `BroadcastJoinLayer` object does not work when `left_on` and/or `right_on` are lists. The simple fix used here is to always covert these arguments from lists to tuples.

- [x] Closes #9870
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Review by ayushdg — APPROVED** [review]
Thanks for the fix! I've confirmed this fixes the repro in the issue as well as my workflow usecase.

**Comment by j-bennet** [comment]
@rjzamora This looks good, are you waiting on something to merge?

**Comment by rjzamora** [comment]
Thanks for taking a look @ayushdg and @j-bennet ! Will merge after CI

**Pull request merged**
